### PR TITLE
Fix Typography panel rendering from style hooks

### LINKED ISSUE #22602: LineHeight styles: Typography panel still visible after disabling
After disabling the new `line-height` style via...

```
add_theme_support( 'disable-custom-line-height' );
```

The Typography panel on the side renders as empty:

82829313-47410700-9eb3-11ea-89b7-7172dfd1bb14

**Expected behavior**

If line-height is disabled, and if the typography panel is empty, it should not render.

Reported by: @mariohamann 

Originally mentioned:
https://github.com/WordPress/gutenberg/issues/20339

### PR DESCRIPTION
**Pull request by ItsJonQ**
This update fixes the Typography sidebar panel rendering, specifically if line-height/font-size style properties are disabled.

The Typography panel will no longer render if it does not contain any style controls.

## How has this been tested?

This has been tested locally within Gutenberg.

* Run `npm run dev`
* Add `add_theme_support( 'disable-custom-line-height' )` to your theme's `function.php`
* Add a Paragraph block
* Typography panel should still render, but without line height controls

* Remove font-sizes for Paragraph and/or Heading block
* Refresh
* Ensure that there is no Typography panel

## Screenshots

Screen Shot 2020-05-25 at 12 35 47 PM

## Types of changes

* Create a Typography panel component under `hooks/` that handles the rendering of typography style controls

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. 

Resolves: https://github.com/WordPress/gutenberg/issues/22602

### CODE REVIEW

**Review by jorgefilipecosta — COMMENTED** [review]
Hi @ItsJonQ,
I used the following code to disable font size and line-height picking:
```
function disable_typography() {
	add_theme_support( 'disable-custom-line-height' );
	add_theme_support( 'disable-custom-font-sizes' );
	add_theme_support( 'editor-font-sizes', array() );
}
add_action( 'after_setup_theme', 'disable_typography' );
```

But the Typography panel still appeared (empty). I guess the panel should not appear at all. Do we have a bug, or I'm missing something in my tests?

File: packages/block-editor/src/hooks/typography.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		hasBlockSupport( blockName, key )
+	);
+
+	const contentMarkup = (
+		<>
+			<FontSizeEdit { ...props } />
+			<LineHeightEdit { ...props } />
+		</>
+	);
+	const hasContent = !! contentMarkup;
```
Comment (youknowriad) [inline packages/block-editor/src/hooks/typography.js:34]: Would this be always true?

**Comment by ItsJonQ** [comment]
@youknowriad + @jorgefilipecosta Thanks for the feedback! I'm currently attempting to test + fix. My local 🐳 Docker environment is having some problems 🙃 

Will update as soon as I can!

File: packages/block-editor/src/hooks/typography.js (hunk 1 above)
Comment (ItsJonQ) [reply packages/block-editor/src/hooks/typography.js:34]: 🤦 Indeed. I think I had a false positive in my testing due to local Docker issues

Thanks for the heads up!

**Comment by ItsJonQ** [comment]
(I was able to get Docker behaving again)

@jorgefilipecosta I had a quick question re: `add_theme_support( 'editor-font-sizes', array() );`

It doesn't seem to work for me 🤔 

I'm getting an array of the 5 default sizes when I do this:

```js
	const { fontSizes } = useSelect( ( select ) =>
		select( 'core/block-editor' ).getSettings()
	);
```

---

As an aside... Originally, I wanted the TypographyPanel to be less involved in tracking either its internal elements need to appear or not (and render accordingly).

However, based on deeper exploration, I'm not sure if that can happen.

To hide the panel, I think the TypographyPanel needs to check...

* if `settings.fontSizes` is empty
* if `settings.disableCustomFontSizes` is `true`
* if `settings. __experimentalDisableCustomLineHeight` is `true`
* if `LINE_HEIGHT_SUPPORT_KEY` and `FONT_SIZE_SUPPORT_KEY` are disabled.

**Comment by jorgefilipecosta** [comment]
> @jorgefilipecosta I had a quick question re: add_theme_support( 'editor-font-sizes', array() );

> It doesn't seem to work for me 🤔


Could you search for editor-font-sizes in your theme folder? Maybe your theme is defining a set of font sizes in some place of the code?

**Comment by youknowriad** [comment]
In the "mu-plugins" loaded in the default wp-env, there's one plugin that resets  the colors and font sizes I think.

**Comment by ItsJonQ** [comment]
@jorgefilipecosta  + @youknowriad Hmm 🤔 

I tried using an older theme (Twenty Eleven). I've added the disable hook at the end of the `functions.php`. The font sizes are still showing up (both in the `useSelect` data and in the UI)

Screen Shot 2020-05-28 at 3 41 27 PM

> In the "mu-plugins" loaded in the default wp-env

Screen Shot 2020-05-28 at 3 42 14 PM

These are the only plugins I can see that are active 😊 . How may I locate this `mu-plugin` and/or adjust the color and font settings?

(Apologies in advance for the hassle 🙏 )

**Comment by youknowriad** [comment]
@ItsJonQ There's a "Must-use" link at the top of your screenshot, you'll see them there. Or alternatively, there's a file  called `normalize-theme.php` on the repo, comment the function calls there.

**Comment by ItsJonQ** [comment]
@youknowriad Ah! Thank you 🙏

**Comment by ItsJonQ** [comment]
@youknowriad + @jorgefilipecosta Haiii :D

I just pushed up a change which **I think** should work. However, I'm still having trouble with the `add_theme_support( 'editor-font-sizes', array() );` config.

I can't seem to make the `fontSizes` setting empty.

I located the `normalize-theme.php` file (wow! super handy), and was able to add my settings there.

I was able to disable line height + custom font sizes. However, I'm not able to remove (or even add) font sizes 🤔 

For example, this doesn't seem to work for me:


```php
add_theme_support( 'editor-font-sizes', array(
	array(
		'name' => 'Small',
		'size' => 12,
		'slug' => 'small'
	),
) );
```

Any insight would be helpful 🙏 !
Thank you!

**Comment by youknowriad** [comment]
Sorry wrong click.

So, now the link control is gone but the typography panel remains empty.

**Comment by ItsJonQ** [comment]
@youknowriad Thank you for re-checking! I was able to test and get things working. I pushed up a fix.

Previously, I think I was getting false results because of my Docker environment. I reset some things this week, and it seemed be working okay.

**Review by youknowriad — APPROVED** [review]
This works great thanks for the fixes.

**Comment by ItsJonQ** [comment]
@youknowriad Thank you 🙏

**Pull request merged**